ci/tests/vmachine/vmachine_test.py

```diff
@@ -184,3 +223,86 @@ def vms_with_fio_test():
             counter = 0
     vpool = api.get_component_by_name('vpools', VPOOL_NAME)[0]
     assert len(vpool['vdisks_guids']) == 0, "Still some disks left on the vpool after waiting {0} seconds: {1}".format(TIMEOUT, vpool['vdisks_guids'])
+
+
+def set_vm_as_template_test():
+    """
+    {0}
+    """.format(general.get_function_name())
+
+    general.check_prereqs(testcase_number=2,
+                          tests_to_run=testsToRun)
+
+    api = Connection.get_connection()
+    vpool_list = api.get_component_by_name('vpools', VPOOL_NAME)
+    assert len(vpool_list), "No vpool found where one was expected"
+    vpool = vpool_list[0]
+
+    create_raw_disk(vpool['name'], TEMPLATE_DISK_NAME)
+    create_machine_from_existing_raw_disk(TEMPLATE_MACHINE_NAME, vpool['name'], TEMPLATE_DISK_NAME)
+
+    counter = TIMEOUT / TIMER_STEP
```

Comment:
This code which waits for the machine to turn up, should go in create_machine_from_existing_raw_disk
and perhaps be parameterized with a wait flag
If you want to wait for it to turn up: wait=True  else wait=False